src/client/src/apps/MainRoute/widgets/blotter/BlotterContainer.tsx

```diff
@@ -7,15 +7,53 @@ import { BlotterActions } from './actions'
 import Blotter from './components'
 import { selectBlotterRows, selectBlotterStatus } from './selectors'
 import { usePlatform } from 'rt-platforms'
+import { Trade } from 'rt-types'
+
+export type FieldValues = ReadonlyArray<any> | undefined;
+export type BlotterFilter = { [fieldId: string]: FieldValues }
 
 interface BlotterContainerOwnProps {
+  filter?: BlotterFilter
   onPopoutClick?: () => void
   tornOff?: boolean
   tearable?: boolean
 }
 
-const mapStateToProps = (state: GlobalState) => ({
-  rows: selectBlotterRows(state),
+const tradeMatchesFilter = (trade: Trade, filterField: string, filteringFieldValues: FieldValues) => {
+  if (!trade) {
+    return false
+  }
+  if (!filteringFieldValues) {
+    return true
+  }
+  if (!(trade as any).hasOwnProperty(filterField)) {
+    console.warn(`Trying to filter of field ${filterField} which does not exist in 'Trade' object`);
+    return true
+  }
+
+  const tradeFieldValue = trade[filterField]
+
+  return filteringFieldValues.includes(tradeFieldValue)
+}
+
+function selectBlotterRowsAndFilter(state: GlobalState, filters?: BlotterFilter) {
+  const trades: ReadonlyArray<Trade> = selectBlotterRows(state)
+  return trades.filter(
+    trade => {
+      if (!filters) {
+        return true
+      }
+      const fieldsToFilterBy = Object.keys(filters);
+      if (fieldsToFilterBy.length === 0) {
```

Comment:
would move these checks outside of the loop

```
function selectBlotterRowsAndFilter(state: GlobalState, filters?: BlotterFilter) {
  const trades: ReadonlyArray<Trade> = selectBlotterRows(state)
  const fieldsToFilterBy = Object.keys(filters || {})
  if (!filters || fieldsToFilterBy.length === 0) {
    return trades;
  }
  
  return trades.filter(
    trade => {
      return fieldsToFilterBy.every(fieldToFilterBy => tradeMatchesFilter(trade, fieldToFilterBy, filters[fieldToFilterBy]))
    }
  )
}
```

Reply:
done, great suggestion